R/ms_graph.R

```diff
@@ -260,11 +260,88 @@ public=list(
         self$get_group(group_id)$delete(confirm=confirm)
     },
 
+    search_site=function(site_name)
+    {
+        op <- sprintf("sites?search=%s", site_name)
+
+        az_site$new(self$token, self$tenant, self$call_graph_endpoint(op))
+    },
+
+    get_site=function(site_id)
+    {
+        op <- file.path("sites", curl::curl_escape(site_id))
+
+        az_site$new(self$token, self$tenant, self$call_graph_endpoint(op))
+    },
+
+    get_drive=function(drive_id)
+    {
+        op <- file.path("drives",curl::curl_escape(drive_id))
+
+        az_drive$new(self$token, self$tenant, self$call_graph_endpoint(op))
+    },
+    
+    get_file=function(drive_id, file_name)
+    {
+        op <- sprintf("/drives/%s/items/root:/%s:", curl::curl_escape(drive_id), curl::curl_escape(file_name))
+
+        az_file$new(self$token, self$tenant, self$call_graph_endpoint(op))
+    },
+
+    download_file=function(file, localfilePath) 
+    {
+        download.file(file$properties["@microsoft.graph.downloadUrl"], localfilePath)
```

Comment:
`httr::GET` may be better than `download.file`

Reply:
ok, i'll adjust to using httr:GET